command/agent/agent.go

```diff
@@ -777,14 +777,34 @@ func (a *Agent) Stats() map[string]map[string]string {
 
 // ShouldReload determines if we should reload the configuration and agent
 // connections. If the TLS Configuration has not changed, we shouldn't reload.
-func (a *Agent) ShouldReload(newConfig *Config) (bool, bool) {
+func (a *Agent) ShouldReload(newConfig *Config) (bool, bool, bool) {
+	var shouldReloadAgent, shouldReloadHTTP, shouldReloadRPC bool
+
 	a.configLock.Lock()
 	defer a.configLock.Unlock()
-	if a.config.TLSConfig.Equals(newConfig.TLSConfig) {
-		return false, false
+
+	var tlsInfoChanged bool
+	if !a.config.TLSConfig.CertificateInfoIsEqual(newConfig.TLSConfig) {
+		tlsInfoChanged = true
```

Comment:
Would it be easier to `return true, true, true` here  and then you can remove the `tlsInfoChanged` variable